app/containers/ChatContainer/saga.js

```diff
@@ -0,0 +1,29 @@
+import { all, call, takeEvery, take, put } from 'redux-saga/effects';
+import { eventChannel } from 'redux-saga';
+import ws from 'services/websocket';
+import { SEND_MESSAGE_ACTION } from './constants';
+import { receiveMessageAction } from './actions';
+
+
+const sendMessage = ({ payload }) => {
+  ws.send(payload);
+};
+
+const watchMessages = eventChannel((emit) => {
+  ws.onmessage = (evt) => emit(evt);
+  return () => console.log('unsubscribe from ws');
+});
+
+export default function* chatContainerSaga() {
+  yield all([
```

Comment:
We're merging two sagas here. The first is going to call the `sendMessage` function for every SEND_MESSAGE_ACTION it intercepts. 

The 2nd will start an `eventChannel` and proxy every message the channel receives through to the store as a `receiveMessageAction`.

Reply:
this part is probably the hardest to grok

Reply:
Yeah, this is new to me too. Happy to walk through it piece by piece in a call or something